unit_tests/test_base_version.py

```diff
@@ -28,99 +34,119 @@ def general_test(scylla_repo='', linux_distro='', cloud_provider=None):
-        scylla_repo = self.url_base + '-enterprise/enterprise-2024.2/deb/unified/latest/scylladb-2024.2/scylla.list'
-        linux_distro = 'ubuntu-focal'
-        version_list = general_test(scylla_repo, linux_distro)
-        assert set(version_list) == {'6.0', '2024.1', '2024.2'}
-
-    def test_2025_1_dev(self):
-        scylla_repo = self.url_base + '/master/rpm/centos/2025-01-15T09:25:01Z/scylla.repo'
-        linux_distro = 'centos'
-        version_list = general_test(scylla_repo, linux_distro)
-        assert set(version_list) == {'6.2'}
-
-    def test_2025_1_release_ubuntu(self):
-        scylla_repo = self.url_base + '/branch-2025.1/deb/unified/2025-02-16T22:46:42Z/scylladb-2025.1/scylla.list'
-        linux_distro = 'ubuntu-focal'
-        version_list = general_test(scylla_repo, linux_distro)
-        assert {'2024.1', '2024.2', '6.2'}.issubset(set(version_list))
-
-    def test_2025_1_release_centos(self):
-        scylla_repo = self.url_base + '/branch-2025.1/rpm/centos/2025-02-23T16:19:08Z/scylla.repo'
-        linux_distro = 'centos-9'
-        version_list = general_test(scylla_repo, linux_distro)
-        assert {'2024.1', '2024.2', '6.2'}.issubset(set(version_list))
-
-
-if __name__ == "__main__":
-    unittest.main()
+download_url_base = 'http://downloads.scylladb.com'
+url_base = f'{download_url_base}/unstable/scylla'
+
+
+def test_master():
+    """
+    Test that master branch select on specific version for upgrade.
+    (not hardcoding the version, since it keep changing)
+    """
+    scylla_repo = url_base + '/master/rpm/centos/2021-08-29T00:58:58Z/scylla.repo'
+    linux_distro = 'centos'
+    version_list = general_test(scylla_repo, linux_distro)
+    assert len(version_list) == 1
+    assert ComparableScyllaVersion(version_list[0]) >= '2025.1'
+
+
+def test_ubuntu_22_azure_2023_1():
+    """
+    Test that azure backend is returning the correct versions
+    """
+    scylla_repo = url_base +\
+        '-enterprise/enterprise-2023.1/deb/unified/2023-05-17T23:03:35Z/scylladb-2023.1/scylla.list'
+    linux_distro = 'ubuntu-jammy'
+    cloud_provider = 'azure'
+    version_list = general_test(scylla_repo, linux_distro, cloud_provider)
+    assert set(version_list) == {'5.2', '2023.1'}
+
+
+def test_2021_1():
+    """ Test that development branch of 2021.1 is returned correct versions """
+    scylla_repo = url_base + '-enterprise/branch-2021.1/rpm/centos/2021-08-29T00:58:58Z/scylla.repo'
+    linux_distro = 'centos'
+    version_list = general_test(scylla_repo, linux_distro)
+    assert set(version_list) == {'4.3', '2020.1', '2021.1'}
+
+
+def test_2021_1_with_centos8_rpm():
+    """ Test that development branch of 2021.1 with centos8 is returned correct versions """
+    scylla_repo = url_base + '-enterprise/branch-2021.1/rpm/centos/2021-08-29T00:58:58Z/scylla.repo'
+    linux_distro = 'centos-8'
+    version_list = general_test(scylla_repo, linux_distro)
+    assert set(version_list) == {'4.3', '2021.1'}
```

Comment:
Do we really need tests for such old versions?

Reply:
they help testing the logic is correct, even if it's old release.
I don't thin we don't really need to add tests for each release....

Reply:
I've removed older unsupported version checks